### PR TITLE
LABS-463 add functionality to read and set windowid field

### PR DESCRIPTION
**Pull request by Z-Shang**

### CODE REVIEW

**Comment by steve-1820** [comment]
LGTM

**Review by steve-1820 — APPROVED** [review]

**Review by steve-1820 — APPROVED** [review]

File: src/windowing/document_source_coprime_timed_slide_window.cpp [hunk 1]
```diff
@@ -180,6 +180,9 @@ DocumentSourceCoprimeTSWindowImpl<TimeRatio>::doGetNext() {
       _state = POP;
       MutableDocument event;
       event.metadata().setEventTime(timestamp);
+      event.metadata().setWindowId(
+          Value(getStageId().toString() + "-" + std::to_string(_totalWindow)));
```
Comment (steve-1820) [inline src/windowing/document_source_coprime_timed_slide_window.cpp:184]: @Z-Shang Do we use this for concurrency tests (i.e. for the composite key during $group)? If so, how does this work during a restart when _totalWindow will be reset?

**Review by steve-1820 — CHANGES_REQUESTED** [review]

File: src/windowing/document_source_coprime_timed_slide_window.cpp (hunk 1 above)
Comment (Z-Shang) [reply src/windowing/document_source_coprime_timed_slide_window.cpp:184]: good point, let me concat an additional timestamp to prevent this from happening

**Review by michaelcahill — APPROVED** [review]

File: src/windowing/document_source_coprime_timed_slide_window.cpp (hunk 1 above)
Comment (steve-1820) [reply src/windowing/document_source_coprime_timed_slide_window.cpp]: Cool, thanks for changing @Z-Shang LGTM

**Review by steve-1820 — APPROVED** [review]

**Pull request merged**